src/game_objects/grid.cpp

```diff
@@ -60,8 +53,26 @@ void Grid::setup_listeners() {
         events::build::request_build_event(c);
     });
 
-    events::build::construct_changed_event.connect([&](ConstructType type) {
+    events::build::construct_selected_event.connect([&](ConstructType type) {
         selected = type;
+
+        // Set preview construct accordingly
+        switch(type) {
```

Comment:
I was working with the confines of what was originally structurally present, i.e. only a ConstructType was provided to me. To do otherwise would be to change other parts of grid. What do you suggest I do? Ideally I would want to redo Grid.